### PR TITLE
bmcweb: simics core dumps

### PR DESCRIPTION
**Pull request by sbteeks**
This commit cherry picks the following upstream commits to handle a
timer system using steady_timer instead of using a timer queue setup:
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/49318
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/49380
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/49382
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/49988

Also, the connection timeout is set to 26 minutes to logged in users
to support our large code update images, otherwise timeout is set to
120 seconds.

Testing:
- Validator test passed.
- Certificate upload did not cause core dump, was tried several hundreds
  of times.
- Code update looks good in terms of performance as compared to
  current implementation.

### CODE REVIEW

**Comment by gtmills** [comment]
> License upload did not cause core dump, was tried several hundreds
of times.

This is certificate upload/replace right? License is something else in Redfish.

File: http/http_connection.hpp [hunk 1]
```diff
@@ -716,64 +669,54 @@ class Connection :
...
+        timer.cancel();
     }
 
-    void startDeadline(size_t timerIterations)
+    void startDeadline()
     {
         cancelDeadlineTimer();
 
-        if (timerIterations)
+        std::chrono::seconds timeout(120);
```
Comment (gtmills) [inline http/http_connection.hpp:679]: @joseph-reynolds  @anoo1 How do you feel about these numbers? 26 mins for users logged in (like from the GUI). 
2 mins for users not logged in. We could go 26mins/26 mins, but I don't see a use case.

**Comment by sbteeks** [comment]
> > License upload did not cause core dump, was tried several hundreds
> > of times.
> 
> This is certificate upload/replace right? License is something else in Redfish.

Yes, Certificate update. Will change the commit message!

**Review by gtmills — APPROVED** [review]

**Comment by gtmills** [comment]
This should fix 
https://w3.rchland.ibm.com/projects/bestquest/?defect=SW554588
https://w3.rchland.ibm.com/projects/bestquest/?defect=SW550264

**Pull request merged**